scylla.go

```diff
@@ -365,7 +368,30 @@ func (p *scyllaConnPicker) Pick(t token) *Conn {
 		return nil
 	}
 
-	idx := p.shardOf(mmt)
+	idx := -1
+
+	p.conns[0].mu.Lock()
```

Comment:
@sylwiaszunejko This code seems to assume that `p.conns[0]` is never `nil`. I don't think this is always true. AFAIK `p.conns` is indexed by the shard of the connection. It may temporarily happen that there is no connection to shard 0 for some time (e.g. the first connection to the node landed on a different shard than 0).

I think this might be the cause of https://github.com/scylladb/scylla-bench/issues/134. You can try reproducing it by writing with a high concurrency and shutting down a node or starting one. The node should have more than one shard set up (the more, the easier to reproduce).

Reply:
You are probably right about the root cause of this issue, in this part of the code I needed to take any connection just to check if `tabletsRoutingV1` is true, but taking `p.conns[0]` was definitely not the right approach to do that. I will try to confirm that it is true and fix it. Thank you for your feedback

Reply:
@piodul created PR to fix this comment: https://github.com/scylladb/gocql/pull/158